src/test/frameworks/compose/discover/runtime/node.spec.ts

```diff
@@ -0,0 +1,322 @@
+import { MockFileSystem } from "../mockFileSystem";
+import { expect } from "chai";
+import {
+  NodejsRuntime,
+  PackageJSON,
+} from "../../../../../frameworks/compose/discover/runtime/node";
+import { FrameworkSpec } from "../../../../../frameworks/compose/discover/types";
+
+describe("NodejsRuntime", () => {
+  let nodeJSRuntime: NodejsRuntime;
+
+  before(() => {
+    nodeJSRuntime = new NodejsRuntime();
+  });
+
+  describe("getNodeImage", () => {
+    it("should return a valid node Image", () => {
+      const version: Record<string, string> = {
+        node: "18",
+      };
+      const actualImage = nodeJSRuntime.getNodeImage(version);
+      const expectedImage = "node:18-slim";
+
+      expect(actualImage).to.deep.equal(expectedImage);
+    });
+
+    it("should return node Image", () => {
+      const version: Record<string, string> = {
+        node: "16",
+      };
+      const actualImage = nodeJSRuntime.getNodeImage(version);
+      const expectedImage = "node:16-slim";
+
+      expect(actualImage).to.deep.equal(expectedImage);
+    });
+  });
+
+  describe("getPackageManager", () => {
+    it("should return yarn package manager", async () => {
+      const fileSystem = new MockFileSystem({
+        "yarn.lock": "It is test file",
+      });
+      const actual = await nodeJSRuntime.getPackageManager(fileSystem);
+      const expected = "yarn";
+
+      expect(actual).to.equal(expected);
+    });
+  });
+
+  describe("getDependenciesNPM", () => {
+    it("should return direct and transitive dependencies", async () => {
+      const fileSystem = new MockFileSystem({
+        "package.json": JSON.stringify({
+          dependencies: {
+            express: "^4.18.2",
+          },
+          devDependencies: {
+            nodemon: "^2.0.12",
+            mocha: "^9.1.1",
+          },
+        }),
+        "package-lock.json": JSON.stringify({
+          packages: {
+            "node_modules/express": {
+              dependencies: {
+                accepts: "~1.3.8",
+                "array-flatten": "1.1.1",
+              },
+            },
+            "node_modules/nodemon": {
+              dependencies: {
+                chokidar: "^3.5.2",
+                debug: "^3.2.7",
+              },
+            },
+            "node_modules/mocha": {
+              dependencies: {
+                "escape-string-regexp": "4.0.0",
+                "find-up": "5.0.0",
+              },
+            },
+          },
+        }),
+      });
+      const packageJSON: PackageJSON = {
+        dependencies: {
+          express: "^4.18.2",
+        },
+        devDependencies: {
+          nodemon: "^2.0.12",
+          mocha: "^9.1.1",
```

Comment:
nit* looks like this can be extracted as a common variable to use in both MockFS and here.

Reply:
Moved some common attributed to before() method.